crt_portal/cts_forms/views.py

```diff
@@ -182,6 +183,38 @@ def _related_reports_count(report):
         return 1
 
 
+def _get_email(report):
+    return report.contact_email
+
+
+def _set_constant_writer(contact_templates, report):
+    for contact_template in contact_templates:
+        if contact_template[1].upper() == report.contact_email.upper():
+            report.constant_writer = True
+
+
+def _was_sent_constant_writer_email(reports):
+    report_emails = tuple(map(_get_email, reports))
+    if len(report_emails) == 0:
+        return
+    with connection.cursor() as cursor:
+        try:
+            cursor.execute(
```

Comment:
performance question: did you `EXPLAIN` this query to confirm it's leveraging your new index?

Reply:
I did and it wasn't using it but I think that's because I don't have enough data on local to make it worth it to use the index.

Reply:
Here's what I'm seeing on local:

- 🏁 it appears to be using the description index
- 🐌 but it is taking a bit to scan the report emails for the `contact_email IN` filter

My suggestion: Add an additional index on email (which I imagine won't just be used here) seems to speed things up a bit:

```sql
CREATE INDEX IF NOT EXISTS cts_forms_report_contact_email_idx
    ON public.cts_forms_report USING btree
    (contact_email)
```

---

Proofs:

Without email index:

```sql
 Sort  (cost=1558.61..1558.85 rows=99 width=109)
   Sort Key: act."timestamp" DESC
   ->  Nested Loop  (cost=0.41..1555.32 rows=99 width=109)
         ->  Seq Scan on cts_forms_report r  (cost=0.00..1074.08 rows=16 width=22)
               Filter: ((contact_email)::text = ANY ('{caitlynboyd@example.com,loriglover@example.com,millersandra@example.com,ybanks@example.com,martinezrobert@example.com,ugarcia@example.com,stephanie86@example.com,jhawkins@example.com,williamsims@example.com,richardsonkeith@example.com,amycopeland@example.com,phogan@example.com,jenniferfox@example.com,pruittmelissa@example.com,richardcarroll@example.com}'::text[]))
         ->  Index Scan using actstream_action_description_idx on actstream_action act  (cost=0.41..30.02 rows=6 width=87)
               Index Cond: (description = concat('Email sent: ''CRT - Constant Writer'' to ', r.contact_email, ' via govDelivery TMS'))
               Filter: ((verb)::text = 'Contacted complainant:'::text)
```

<img width="1178" alt="image" src="https://user-images.githubusercontent.com/15126660/219385201-987ce375-3dd6-441e-a687-2b0944526bd1.png">

With email index:

```sql
 Sort  (cost=545.03..545.28 rows=99 width=109)
   Sort Key: act."timestamp" DESC
   ->  Nested Loop  (cost=0.70..541.75 rows=99 width=109)
         ->  Index Only Scan using cts_forms_report_contact_email_idx on cts_forms_report r  (cost=0.28..60.51 rows=16 width=22)
               Index Cond: (contact_email = ANY ('{caitlynboyd@example.com,loriglover@example.com,millersandra@example.com,ybanks@example.com,martinezrobert@example.com,ugarcia@example.com,stephanie86@example.com,jhawkins@example.com,williamsims@example.com,richardsonkeith@example.com,amycopeland@example.com,phogan@example.com,jenniferfox@example.com,pruittmelissa@example.com,richardcarroll@example.com}'::text[]))
         ->  Index Scan using actstream_action_description_idx on actstream_action act  (cost=0.41..30.02 rows=6 width=87)
               Index Cond: (description = concat('Email sent: ''CRT - Constant Writer'' to ', r.contact_email, ' via govDelivery TMS'))
               Filter: ((verb)::text = 'Contacted complainant:'::text)
(8 rows)
```

<img width="1249" alt="image" src="https://user-images.githubusercontent.com/15126660/219385361-95a441ad-df86-4adc-92af-46cc040f6b69.png">